data/actions/scripts/other/teleport.lua

```diff
@@ -1,5 +1,10 @@
 local upFloorIds = {1386, 3678, 5543}
 function onUse(player, item, fromPosition, target, toPosition, isHotkey)
+	local tile = item:getTile()
+	if tile and tile:hasFlag(TILESTATE_PROTECTIONZONE) and player:isPzLocked() then
```

Comment:
You should also check destination tile.
There should also be a check on the rope script side to prevent players from roping others that are PZ locked to PZ zone.